### ISSUE TITLE
blank preview

### ISSUE THREAD

**Issue by Vtin** [fileability/chocolat-public#1751]
I've reported this long ago, but as the app has been updated many times and the problem persists, I include it again, for logging purposes.
Chocolat's html preview doesn't work for me. I've worked with it for five minutes, and then the preview window got blank, no matter the changes I made in the document nor the times I refreshed it.
Of course, I've checked the document for errors and previewed it with other apps, with no trouble.
This happens with all the html documents I open with Chocolat.
OS X Yosemite, mac mini (mid 2011).
Thanks;
Val

**Comment by atg** [comment]
Hm, I haven't experienced this issue myself, but perhaps not surprising as I haven't really been using Web Preview that much.

If it's going blank, it's probably got detached from the document. I can imagine a lot of ways that could happen, especially if you're using a detached window. The whole thing needs a rethink really.

**Comment by Vtin** [comment]
Can't help much in this, as I'm not using Chocolat for some time becouse of it (sorry). All I can say is that I cannot preview any html document properly. Some of them previews correctly for the first time, but, afterwards, doesn't accept any change made in the code; others simply doesn't preview at all. Split preview or windowed preview doesn't make much difference, and it happens with all html documents, created by chocolat or not.
Updated to el Capitan recently, no change with this.

**Comment by atg** [comment]
Yeah, I mean personally I hardly ever use Web Preview and just use Chrome, which is an indictment of the feature!

So it needs an overhaul. We need to find a way to work web preview (and project find) into the main editor GUI so they don't get so neglected. I would much prefer if Web Preview showed up in Active Files (or at the bottom of Active Files). Instead of being special it should be a view like any other. That requires pinned tabs though.

Thinking aloud: A good idea might be have two classes of views: files (that show up on the left) and non-files (that show up on the right). This solves the issue of having two files split horizontally and web preview vertically -- which we couldn't do if Web Preview were handled exactly like a file.

That would be nice because then you could just double click web preview in the source list to open it in a new window (like any other file).

It would also be nice to have a whole menu dedicated to Web Preview.

But to answer the main point of this issue, a blank view probably means that Web Preview is detached. it has forgotten which document it is associated with. Currently the UI provides no control over which document is currently being previewed. We need a popup button or something.
